### PR TITLE
Fix for transition dipole error and testing

### PR DESCRIPTION
**Pull request by bhourahine**
Fix for #535

### CODE REVIEW

**Comment by bhourahine** [comment]
The transition dipole vector isn't portable as its directly proportional to the eigenvector, so caries an arbitrary phase

File: prog/dftb+/lib_timedep/linrespgrad.F90 [hunk 1]
```diff
@@ -2034,6 +2035,10 @@ subroutine writeExcitations(sym, osz, nexc, nmatup, getij, win, eval, evec, wij,
       if (.not.tDegenerate) then
         call taggedWriter%write(fdTagged, tagLabels%excEgy, eval)
         call taggedWriter%write(fdTagged, tagLabels%excOsc, osz)
+        if (fdTradip > 0) then
```
Comment (aradi) [inline prog/dftb+/lib_timedep/linrespgrad.F90:2038]: That's a risky check. If units are auto-generated (with ``newunit=``), they will be negative according to the standard.

**Comment by aradi** [comment]
@thomas-niehaus Do you have any ideas how to fix the non-portability of the transition dipole vectors?

File: prog/dftb+/lib_timedep/linrespgrad.F90 (hunk 1 above)
Comment (bhourahine) [reply prog/dftb+/lib_timedep/linrespgrad.F90:2038]: Yes, the whole file and linresp.F90 both need to be refactored to remove generated file id as these are used for _all_ output files.

Comment (aradi) [reply prog/dftb+/lib_timedep/linrespgrad.F90:2038]: OK, I see, it's all over in the file :disappointed:

**Pull request merged**